src/Poseidon/Package.hs

```diff
@@ -149,50 +189,118 @@ readPoseidonPackage yamlPath = do
 findPoseidonPackages :: FilePath -- ^ the base directory to search from
                      -> IO [PoseidonPackage] -- ^ the returned list of poseidon packages.
 findPoseidonPackages baseDir = do
-    entries <- listDirectory baseDir
-    posPac  <- mapM tryReadPoseidonPackage . map (baseDir </>) . filter ((=="POSEIDON.yml") . takeFileName) $ entries
-    forM_ (lefts posPac) $ (\e -> case e of
+    paths <- findAllPOSEIDONymlFiles baseDir
+    allPackages <- findPoseidonPackagesFromPosList paths
+    return $ map snd allPackages
+
+findPoseidonPackagesFromPosList :: [FilePath] -> IO [(FilePath, PoseidonPackage)]
+findPoseidonPackagesFromPosList paths = do
+    pacs <- mapM tryReadPoseidonPackage paths
+    let pacsPacWithPaths = zip paths pacs
+    -- filter out broken packages
+    forM_ (lefts $ map snd pacsPacWithPaths) $ (\e -> case e of
         PoseidonYamlParseException fp err ->
             putStrLn ("Can't read package at " ++ fp ++ " due to YAML parsing error: " ++ show err)
         _ -> error "this should never happen")
-    subDirs     <- filterM doesDirectoryExist . map (baseDir </>) $ entries
-    morePosPacs <- fmap concat . mapM findPoseidonPackages $ subDirs
-    return $ (rights posPac) ++ morePosPacs
+    return $ uncurry zip $ second rights $ unzip $ filter (\(_, a) -> isRight a) pacsPacWithPaths
   where
     tryReadPoseidonPackage :: FilePath -> IO (Either PoseidonException PoseidonPackage)
     tryReadPoseidonPackage = try . readPoseidonPackage
 
+findAllPOSEIDONymlFiles :: FilePath -> IO [FilePath]
```

Comment:
I think we need to clean this up. There is a zoo of `loadXYZ`-related functions now...